lib/main.js

```diff
@@ -0,0 +1,290 @@
+import isPromise from 'is-promise';
+
+// ## printMaxListenersWarning
+// is a listener that prints a warning when the max listeners has been exceeded
+// just like the ``EventEmitter`` does, but now you can clear it out by
+// executing ``this.removeAllListeners('maxListenersPassed')`` in your
+// constructor function just after running the ``EventEmitter`` constructor
+// function in your own constructor function.
+function printMaxListenersWarning(eventName, listenerCount) {
+  console.warn(
+    'The Event ' + eventName + ' has exceeded ' + this._maxListeners +
+    ' listeners, currently at ' + listenerCount);
+}
+
+// # EventEmitter
+// Constructor function, a replacement for ``EventEmitter`` that also allows
+// for asynchronous event listeners and allows event listeners to cancel events
+// if they return `false`. Event listeners may be either synchronous like DOM
+// events or asynchronous to better fit a async/await Promise style.  (Because
+// of that, making an event placed inside of a high performance loop is
+// probably a bad idea because the event listeners can introduce an
+// unpredictable delay.) The ``EventEmitter`` constructor function attaches
+// three "please don't touch" properties to itself (or its inheriting object)
+// to keep track of listeners and the maximum number of allowed listeners.
+export function EventEmitter() {
+  this._eventListeners = new Map();
+  this._onceListeners = new Map();
+  this._maxListeners = 10;
+  this.on('maxListenersPassed', printMaxListenersWarning.bind(this));
+}
+
+// ## emitSync
+// Operates just like ``emit`` except all listeners must be synchronous.
```

Comment:
What's with all the backticks?

Reply:
Sticking with the original style. Some people prefer that so single ticks are easier to type or something. Can change if people want. This should all be converted to jsdoc too.

Reply:
Tracking this future release: https://github.com/digitalbazaar/async-node-events/issues/3